planner/core/find_best_task.go

```diff
@@ -1061,7 +1067,7 @@ func (ds *DataSource) getOriginalPhysicalTableScan(prop *property.PhysicalProper
 	// we still need to assume values are uniformly distributed. For simplicity, we use uniform-assumption
 	// for all columns now, as we do in `deriveStatsByFilter`.
 	ts.stats = ds.tableStats.ScaleByExpectCnt(rowCount)
-	rowSize := ds.TblColHists.GetAvgRowSize(ds.TblCols, false)
+	rowSize := ds.TblColHists.GetTableAvgRowSize(ds.TblCols, ts.StoreType, ds.tableInfo.PKIsHandle)
```

Comment:
For TiFlash scan, we should not use `ds.TblCols` here?

Reply:
Why? I can't get your point.

Reply:
If table has 3 columns `(a, b, c)`, and the query only needs column a, for column-oriented storage, we can just scan column a, instead of the whole row in row-oriented storage?

Reply:
Oh, yes. That my fault.